File: front/medicine/src/app/shared/models/entity/DosageLog.ts
import { DosageLogStatus } from "../../enums/DosageLogStatus";
import { User } from "./User";
import { DosingFrequencyReminder } from "./DosingFrequencyReminder";


export class DosageLog {
  dosageRecommendationId!: number;
  dosageRecommendation!: DosingFrequencyReminder;
  status!: DosageLogStatus;
  quantity!: number;
  dateTime!: Date;
  user?: User;
  userId?: number;
  id!: number;
  createdAt?: Date;
}

File: src/Entities/Medicine.Entities/Models/Dosages/DosageLog.cs
using Medicine.Entities.Enums;
using Medicine.Entities.Models.Base;

namespace Medicine.Entities.Models.Dosages
{
    public class DosageLog : EntityUser
    {
        public int DosageRecommendationId { get; set; }
        public DosingFrequencyReminder DosageRecommendation { get; set; }
        public DosageLogStatus Status { get; set; }
        public double Quantity { get; set; }
        public DateTime DateTime { get; set; }
    }
}
